src/idl_gen_rust.cpp

```diff
@@ -2472,12 +2587,45 @@ class RustGenerator : public BaseGenerator {
       code_ += "  }\n";
       // Setter.
       if (IsStruct(field.value.type)) {
-        code_.SetValue("FIELD_SIZE",
-                       NumToString(field.value.type.struct_def->bytesize));
+        code_.SetValue("FIELD_SIZE", NumToString(InlineSize(field.value.type)));
         code_ += "  pub fn set_{{FIELD_NAME}}(&mut self, x: &{{FIELD_TYPE}}) {";
         code_ +=
             "    self.0[{{FIELD_OFFSET}}..{{FIELD_OFFSET}}+{{FIELD_SIZE}}]"
             ".copy_from_slice(&x.0)";
+      } else if (IsArray(field.value.type)) {
+        if (GetFullType(field.value.type) == ftArrayOfBuiltin) {
+          code_.SetValue("ARRAY_ITEM",
+                         GetTypeGet(field.value.type.VectorType()));
+          code_.SetValue(
+              "ARRAY_ITEM_SIZE",
+              NumToString(InlineSize(field.value.type.VectorType())));
+          code_ +=
+              "  pub fn set_{{FIELD_NAME}}(&mut self, items: &{{FIELD_TYPE}}) "
+              "{";
+          code_ += "    for (i, item) in items.iter().enumerate() {";
+          code_ += "      let item_le = item.to_little_endian();";
+          code_ += "      unsafe {";
+          code_ += "        core::ptr::copy_nonoverlapping(";
+          code_ += "          &item_le as *const {{ARRAY_ITEM}} as *const u8,";
+          code_ +=
+              "          self.0[{{FIELD_OFFSET}} + {{ARRAY_ITEM_SIZE}} * "
+              "i..].as_mut_ptr(),";
+          code_ += "          {{ARRAY_ITEM_SIZE}},";
+          code_ += "        );";
+          code_ += "      }";
+          code_ += "    }";
+        } else {
+          code_.SetValue("FIELD_SIZE",
+                         NumToString(InlineSize(field.value.type)));
+          code_ +=
+              "  pub fn set_{{FIELD_NAME}}(&mut self, x: &{{FIELD_TYPE}}) {";
+          code_ +=
+              "    let src = unsafe { ::std::slice::from_raw_parts(x.as_ptr() "
```

Comment:
I think this doesn't work unless the pointer is aligned.

Reply:
I'm using it as a byte pointer, don't think it requires any alignment. Also from what I understand there is a padding as part of the start so that every field will be aligned to 8 bytes. So if we have a byte at offset 64 the next field will start at offset 72.

Reply:
Ah, you're right - I misread the intent. It might be clearer as
```rust
std::ptr::copy(
    x.as_ptr() as *const u8,
    self.0.as_mut_ptr().add(FIELD_OFFSET),
    FIELD_SIZE
);
```